.github/workflows/build.yml

```diff
@@ -0,0 +1,49 @@
+name: Task - Build Madara
+on:
+  workflow_dispatch:
+  workflow_call:
+jobs:
+  build:
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - uses: Swatinem/rust-cache@v2
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.78.0
+      - name: Setup build deps
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y clang llvm libudev-dev protobuf-compiler
+      - uses: rui314/setup-mold@v1
+      - uses: software-mansion/setup-scarb@v1
+        with:
+          scarb-version: "2.8.2"
+      - uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+      - uses: software-mansion/setup-scarb@v1
+        with:
+          scarb-version: "2.8.2"
+      - name: Build Cairo contracts
+        run: |
+          cd cairo
+          scarb build
+      - name: Build
+        run: |
+          cargo build --release --bin madara
+      - name: Cache Madara binary
+        uses: actions/cache@v3
+        with:
+          path: target/release/madara
+          key: ${{ runner.os }}-madara-${{ github.sha }}
```

Comment:
Here it appears that you are generating you key over the commit hash this doesnt invalidate the cache on each commit?

Reply:
I have checked maybe you can use:

`key: ${{ runner.os }}-madara-${{ hashFiles('Cargo.lock') }}`

to generate a key based on the lock changes or take in account even more files

Reply:
resolved